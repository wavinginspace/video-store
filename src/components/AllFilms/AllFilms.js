import React, { Component } from 'react';
import FilmLink from '../FilmLink/FilmLink';
import ApiContext from '../../ApiContext';
import '../FilmLink/FilmLink.scss';
import './AllFilms.scss';

export class AllFilms extends Component {
  static contextType = ApiContext;

  render() {
    const { films } = this.props;

    let numberFilms =
      this.context.films.length === 1
        ? `There is 1 film in your store.`
        : `There are ${this.context.films.length} films in your store.`;

    return (
      <>
        <p className="filmnumber fadeIn">{numberFilms}</p>
        <section className="AllFilms fadeIn box">
          <ul>
            {films
              .sort(function(a, b) {
                if (a.title < b.title) {
                  return -1;
                }
                if (a.title > b.title) {
                  return 1;
                }
                return 0;
              })
              .map(film => (
                <FilmLink key={film.id} film={film} />
              ))}
          </ul>
        </section>
      </>
    );
  }
}

export default AllFilms;
