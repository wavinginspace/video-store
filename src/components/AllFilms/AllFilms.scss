@import '../../styles/_breakpoints.scss';
@import '../../styles/_variables.scss';

.fadeIn {
  animation-name: fadeIn;
  animation-duration: 0.5s;
  animation-fill-mode: forwards;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}

.filmnumber {
  margin-bottom: 1rem;
}


