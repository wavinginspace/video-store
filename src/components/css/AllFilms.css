:root {
  --main: var(--main);
  --nav: #499acc;
  --headergrad: linear-gradient(0deg, #ff7676, #fffd8f);
  --light: #fffd8f;
  --headerog: #ff7676;
}

.filmnumber {
  margin-bottom: 1rem;
}

.fadeIn {
  -webkit-animation-name: fadeIn;
          animation-name: fadeIn;
  -webkit-animation-duration: .5s;
          animation-duration: .5s;
  -webkit-animation-fill-mode: forwards;
          animation-fill-mode: forwards;
}

@-webkit-keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
/*# sourceMappingURL=AllFilms.css.map */