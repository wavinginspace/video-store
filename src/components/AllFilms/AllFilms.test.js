import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router} from 'react-router-dom';
import AllFilms from './AllFilms';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Router><AllFilms films={[]}/></Router>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

// it('renders the UI as expected', () => {
//   const tree = renderer.create(<Router><AllFilms films="[test]"/></Router>).toJSON();
//   expect(tree).toMatchSnapshot();
// });
