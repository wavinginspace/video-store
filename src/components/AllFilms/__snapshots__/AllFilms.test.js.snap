// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <p
    className="filmnumber fadeIn"
  >
    There are 0 films in your store.
  </p>,
  <section
    className="AllFilms fadeIn"
  >
    <div
      className="box"
    >
      <ul />
    </div>
    <a
      className="back-button"
      href="/"
      onClick={[Function]}
    >
      <p
        className="back-button-p"
      >
        Back Home
      </p>
    </a>
  </section>,
]
`;
